app/src/main/java/nl/tudelft/bbw/crawler/ReadCrawlerBlocksQuery.java

```diff
@@ -0,0 +1,134 @@
+package nl.tudelft.bbw.crawler;
+
+import static nl.tudelft.bbw.crawler.BlockDatabase.BLOCKS_TABLE_NAME;
+import static nl.tudelft.bbw.crawler.BlockDatabase.USER_TABLE_NAME;
+import static nl.tudelft.bbw.crawler.BlockDatabase.COLUMNS_CHAIN;
+import static nl.tudelft.bbw.crawler.BlockDatabase.COLUMNS_MEMBER;
+import static nl.tudelft.bbw.crawler.BlockDatabase.JOIN_TABLES;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import nl.tudelft.bbw.blockchain.Block;
+import nl.tudelft.bbw.blockchain.BlockData;
+import nl.tudelft.bbw.blockchain.BlockType;
+import nl.tudelft.bbw.blockchain.Hash;
+import nl.tudelft.bbw.blockchain.TrustValues;
+import nl.tudelft.bbw.blockchain.User;
+
+/**
+ * Class which constructs a query to read the crawled database.
+ */
+public class ReadCrawlerBlocksQuery {
+
+    private static final String NA = "N/A";
+    /**
+     * Column indices for blocks
+     */
+    private static final int INDEX_ID = 0;
+    private static final int INDEX_PUB_KEY = 1;
+    private static final int INDEX_SEQ_NO = 5;
+    private static final int INDEX_PREV_PUB = 3;
+    private static final int INDEX_PREV_HASH_CHAIN = 4;
+    /**
+     * The blocks retrieved from the database
+     */
+    private Map<String, List<Block>> chain;
+
+
+    /**
+     * Constructs a query for the blocks
+     */
+    public ReadCrawlerBlocksQuery() {
+    }
+
+    /**
+     * Retrieve the chain which consists of the crawled blocks
+     *
+     * @return The list of blocks sorted on public key.
+     */
+    public Map<String, List<Block>> getChain() {
+        return chain;
+    }
+
+    /**
+     * Method to go through all the rows of the database using the cursor,
+     * and make a block with the values, and eventually write it into a JSON file.
+     */
+    private void parse(Cursor cursor) {
+        if (cursor.getCount() == 0) {
+            chain = null;
+        } else {
+            chain = new HashMap<>((int) (cursor.getCount() * 1.33));
```

Comment:
When writing this, we got a weird lint error, which only could be fixed by preallocating the slots. I looked at it again and now the error is gone, so will change it!